Essentials/src/com/earth2me/essentials/ReserveEssentials.java

```diff
@@ -0,0 +1,1830 @@
+    @Override
+    public CompletableFuture<Boolean> asyncRemoveHoldings(UUID identifier, BigDecimal amount) {
+        return CompletableFuture.supplyAsync(() -> removeHoldings(identifier, amount));
+    }
+
+    /**
+     * Used to remove funds from an account.
+     * @param identifier The identifier of the account that is associated with this call.
+     * @param amount The amount you wish to remove from this account.
+     * @param world The name of the {@link World} associated with the amount.
+     * @return True if the funds were removed from the account, otherwise false.
+     */
+    @Override
+    public CompletableFuture<Boolean> asyncRemoveHoldings(String identifier, BigDecimal amount, String world) {
+        return CompletableFuture.supplyAsync(() -> removeHoldings(identifier, amount));
+    }
+
+    /**
+     * Used to remove funds from an account.
+     * @param identifier The identifier of the account that is associated with this call.
+     * @param amount The amount you wish to remove from this account.
+     * @param world The name of the {@link World} associated with the amount.
+     * @return True if the funds were removed from the account, otherwise false.
+     */
+    @Override
+    public CompletableFuture<Boolean> asyncRemoveHoldings(UUID identifier, BigDecimal amount, String world) {
+        return CompletableFuture.supplyAsync(() -> removeHoldings(identifier, amount));
+    }
+
+    /**
+     * Used to remove funds from an account.
+     * @param identifier The identifier of the account that is associated with this call.
+     * @param amount The amount you wish to remove from this account.
+     * @param world The name of the {@link World} associated with the amount.
+     * @param currency The {@link Currency} associated with the balance.
+     * @return True if the funds were removed from the account, otherwise false.
+     */
+    @Override
+    public CompletableFuture<Boolean> asyncRemoveHoldings(String identifier, BigDecimal amount, String world, String currency) {
+        return CompletableFuture.supplyAsync(() -> removeHoldings(identifier, amount));
+    }
+
+    /**
+     * Used to remove funds from an account.
+     * @param identifier The identifier of the account that is associated with this call.
+     * @param amount The amount you wish to remove from this account.
+     * @param world The name of the {@link World} associated with the amount.
+     * @param currency The {@link Currency} associated with the balance.
+     * @return True if the funds were removed from the account, otherwise false.
+     */
+    @Override
+    public CompletableFuture<Boolean> asyncRemoveHoldings(UUID identifier, BigDecimal amount, String world, String currency) {
+        return CompletableFuture.supplyAsync(() -> removeHoldings(identifier, amount));
+    }
+
+    /**
+     * Used to determine if a call to the corresponding removeHoldings method would be successful. This method does not
+     * affect an accoun
```

Comment:
This is probably possible to implement, but I don't think there's a clean or efficient way to do it.

On a side note, returning `false` everywhere for failures isn't ideal - I'll post a further comment in more depth below.